Refactor optimization interface

ludoro opened the pull request:
@ChrisRackauckas So right now I have not updated test/optimization, but this is what you had in mind with the refactoring? I basically copied your refactor style on the Sampling stuff. 
If you think this is alright then I can change stuff in the test

ChrisRackauckas commented on src/Optimization.jl:
```diff
@@ -23,14 +28,11 @@ function merit_function(point,w,surr::AbstractSurrogate,s_max,s_min,d_max,d_min,
+SRBF ND
 """
-function SRBF(lb,ub,surr::AbstractSurrogate,maxiters::Int,sample_type::SamplingAlgorithm,num_new_samples::Int,obj::Function)
+function surrogate_optimize(obj::Function,::SRBF,lb,ub,surr::AbstractSurrogate,sample_type::SamplingAlgorithm,maxiters::Int,num_new_samples::Int)
```
After the sampling method, the next two should probably be kwargs with a default?

ChrisRackauckas commented on src/Optimization.jl:
```diff
@@ -1,6 +1,11 @@
 using LinearAlgebra
 using Distributions
 
+abstract type OptimizationAlgorithm end
```
Just make it SurrogateOptimizationAlgorithm so it doesn't clash anywhere

ChrisRackauckas commented on src/Surrogates.jl:
```diff
@@ -14,6 +14,6 @@ include("Optimization.jl")
 export Kriging, RadialBasis, add_point!, current_estimate, std_error_at_point
 export sample, SamplingAlgorithm, GridSample, UniformSample, SobolSample, LatinHypercubeSample, LowDiscrepancySample
-export SRBF,LCBS,EI
+export OptimizationAlgorithm,SRBF,LCB,EI
```
no need to export the abstract type, just the user-facing types.

ludoro commented:
@ChrisRackauckas Refactoring looks good, I addressed your suggestions. 
I think I should also add `dtol = 1e-3*norm(ub-lb)` as a kwargs parameter that the user can toggle with,( I mean just the value `1e-3`) because sometimes I still get singularities, in every papers I have read they say it's not "well tested yet", so I think that maybe a user would want to use a particular method then it should be able to relax the bounds a bit, what do you think?

ChrisRackauckas commented:
You shouldn't relax the bounds because you can assume they are put there due to fundamental problems beyond the bounds. For example, if the parameters are negative in some ODE simulations the solution can explode to infinity.
